lib/cell.rb

```diff
@@ -0,0 +1,49 @@
+class Cell
+
+  attr_reader :coordinate, :empty, :ship, :fired_upon
+
+  def initialize(coordinate)
+    @coordinate = coordinate
+    @empty = true
+    @ship = nil
+    @fired_upon = false
+  end
+
+  def empty?
+    @empty
+  end
+
+  def place_ship(ship)
+    @empty = false
+    @ship = ship
+  end
+
+  def fired_upon?
+    @fired_upon
+  end
+
+  def fire_upon
+    @fired_upon = true
+    if @ship == ship
+      ship.hit
+    end
+  end
+
+  def render(show = false)
+    if @fired_upon == false
+      if show == true && empty? == false
```

Comment:
It is apparently not good form to use nested if statements but I didn't see a way to make this section work otherwise. If you know better, please point me in the right direction!

Reply:
No I had to do this on the last project too. They told me the same thing about avoiding nested `if` statements if possible. I would have done it exactly the same way. On another note I read something about `case` statements and they seemed like a possible alternative to nested `if` statements. Might be worth us looking into them if we find our selfs doing alot of nested stuff.

Reply:
Sounds good. I put a card in our iteration 3 list on Trello to refactor, and made a note to revisit nested if statements in cell class!